core/cas-server-core-tickets/src/test/java/org/apereo/cas/ticket/registry/DefaultTicketRegistryTests.java

```diff
@@ -24,61 +46,148 @@
-    void verifyCountForService() throws Throwable {
-        val service = RegisteredServiceTestUtils.getService(UUID.randomUUID().toString());
-        val registry = getNewTicketRegistry();
-        val user = UUID.randomUUID().toString();
+    @SpringBootTest(classes = LogoutTestConfiguration.class,
+        properties = {
+            "cas.ticket.tgt.core.only-track-most-recent-session=false",
+            "cas.ticket.registry.cleaner.schedule.enabled=false"
+        })
+    @ExtendWith(CasTestExtension.class)
+    @Nested
+    class LogoutTests {
 
-        for (int i = 0; i < 5; i++) {
-            val tgt = new MockTicketGrantingTicket(user);
-            val st = tgt.grantServiceTicket(service, TicketTrackingPolicy.noOp());
-            registry.addTicket(st);
-            registry.updateTicket(tgt);
+        @RepeatedTest(1)
+        void verifyGetExpiredTicketLogoutPerformed(final ConfigurableApplicationContext applicationContext) throws Throwable {
+            val registry = new DefaultTicketRegistry(mock(TicketSerializationManager.class), new DefaultTicketCatalog(),
+                    applicationContext);
+            val originalAuthn = CoreAuthenticationTestUtils.getAuthentication();
+            val ticketGrantingTicket = new TicketGrantingTicketImpl(BaseTicketRegistryTests.TestTicketIdentifiers.generate().ticketGrantingTicketId(),
+                    originalAuthn, new HardTimeoutExpirationPolicy(1));
+            registry.addTicket(ticketGrantingTicket);
+            Thread.sleep(1500);
+            val tgtId = ticketGrantingTicket.getId();
+            val tgt = registry.getTicket(tgtId);
+            assertNull(tgt);
+            val internalTgt = registry.getMapInstance().get(tgtId);
+            assertNull(internalTgt);
+            await().atMost(3, TimeUnit.SECONDS).untilAsserted(() -> assertTrue(LogoutTestConfiguration.LogoutManagerTestConfiguration.nbCalls > 0));
         }
-        val count = registry.countTicketsFor(service);
-        assertEquals(5, count);
     }
 
+    @ImportAutoConfiguration({
+        CasCoreTicketsAutoConfiguration.class,
+        CasCoreUtilAutoConfiguration.class,
+        CasCoreScriptingAutoConfiguration.class,
+        CasPersonDirectoryAutoConfiguration.class,
+        CasCoreLogoutAutoConfiguration.class,
+        CasCoreServicesAutoConfiguration.class,
+        CasCoreAuthenticationAutoConfiguration.class,
+        CasCoreAutoConfiguration.class,
+        CasCoreCookieAutoConfiguration.class,
+        CasCoreWebAutoConfiguration.class,
+        CasCoreWebflowAutoConfiguration.class,
+        CasCoreMultifactorAuthenticationAutoConfiguration.class,
+        CasCoreMultifactorAuthenticationWebflowAutoConfiguration.class,
+        CasCoreNotificationsAutoConfiguration.class,
+        CasCoreEventsAutoConfiguration.class
+    })
+    @SpringBootConfiguration(proxyBeanMethods = false)
+    @SpringBootTestAutoConfigurations
+    public static class LogoutTestConfiguration {
```

Comment:
This should be unnecessary; We have `BaseTicketRegistryTests.SharedTestConfiguration.class`so you only likely need your own logoutManager bean

Reply:
In fact, this is not exactly the same list of imported classes: `CasCoreEventsAutoConfiguration.class` has been added for this test.